handlers/benefit_eligible.go

```diff
@@ -0,0 +1,109 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/DTS-STN/benefit-service/bindings"
+	"github.com/DTS-STN/benefit-service/models"
+	"github.com/DTS-STN/benefit-service/renderings"
+	"github.com/DTS-STN/benefit-service/src/benefits"
+	"github.com/labstack/echo/v4"
+)
+
+func (h *Handler) BenefitsEligibility(c echo.Context) error {
+	var benefitsResponse = new(renderings.BenefitsResponse)
+	var benefit models.Benefits
+	var err error
+
+	//Bind the request into our request struct
+	eligible := new(bindings.BenefitEligibilityRequest)
+	if err = c.Bind(eligible); err != nil {
+		errObj := new(models.Error)
+		errObj.Status = http.StatusBadRequest
+		errObj.ErrorMessage = err.Error()
+		return c.JSON(http.StatusBadRequest, errObj)
+	}
+
+	//Default language is English, so set Lang to en if not otherwise specified
+	if eligible.Lang == "" {
+		eligible.Lang = "en"
+	}
+
+	//Check to see if any of the fields are empty
+	//TODO: Clean this up/reduce redundant code
+	if eligible.IncomeDetails == "" {
+		errObj := new(models.Error)
+		errObj.Status = http.StatusBadRequest
+		errObj.ErrorMessage = "incomeDetails were not supplied"
+		return c.JSON(http.StatusBadRequest, errObj)
+	}
+
+	if eligible.TimeOutOfWork == "" {
+		errObj := new(models.Error)
+		errObj.Status = http.StatusBadRequest
+		errObj.ErrorMessage = "timeOutOfWork was not supplied"
+		return c.JSON(http.StatusBadRequest, errObj)
+	}
+
+	if eligible.AbleToWork == "" {
+		errObj := new(models.Error)
+		errObj.Status = http.StatusBadRequest
+		errObj.ErrorMessage = "ableToWork was not supplied"
+		return c.JSON(http.StatusBadRequest, errObj)
+	}
+
+	if eligible.ReasonForOutOfWork == "" {
+		errObj := new(models.Error)
+		errObj.Status = http.StatusBadRequest
+		errObj.ErrorMessage = "reasonForOutOfWork was not supplied"
+		return c.JSON(http.StatusBadRequest, errObj)
+	}
+
+	if eligible.Gender == "" {
+		errObj := new(models.Error)
+		errObj.Status = http.StatusBadRequest
+		errObj.ErrorMessage = "gender was not supplied"
+		return c.JSON(http.StatusBadRequest, errObj)
+	}
+
+	//Determine if individual qualifies for Regular EI Benefit
+	if (eligible.IncomeDetails != "lt-30k") &&
```

Comment:
I was looking more towards a match function rather than hard-coded if statements.

This would work in the following way.

the match function would take three arguments 

1) data ( this would be the body of the request)
2) data to match 
3) benefit string if data matches 

If the data matches the data to match (keys are the same) then the benefit string is returned. 

you would essentially call the match function for every benefit you have and push the result for each of these into an array which would be returned by the handler